controller/rest/sniffer.go

```diff
@@ -64,6 +64,7 @@ func isSnifferAccessible(id string, acc *access.AccessControl) (string, []*share
 			if !acc.Authorize(&share.CLUSSnifferDummy{WorkloadDomain: wl.Domain}, nil) {
 				status = nil
 				err = common.ErrObjectAccessDenied
+				return agentId, status, err
```

Comment:
why add line 67 which returns the same thing as line 72 when acc.Authorize() returns false?

Reply:
This is due to **variable shadowing** caused by the use of the := operator within the nested if statement.
We have two err declaration in this function.

**Outer err Declaration:**
status, err := rpc.GetSniffers(agentId, f)

**Inner err Declaration:**
if wl, err := cacher.GetWorkloadBrief(status[0].WorkloadID, "", acc)

This inner err will shadows the outer err. 
Any assignments to err within this block affect only the inner err, leaving the outer err unchanged.

Reply:
a simplified code as below..
The number-1 and number-2 err assignment has no effect, no code have chance to use the value.
Because the err will be assigned in number-4 and it will use be used in number-5 for the final return value.

```
func GetCspConfig() api.RESTFedCspSupportResp {
	var err error

	if obj, err = global.ORCH.GetResource(RscTypeConfigMap, NvAdmSvcNamespace, "csp-config"); err == nil {
		if cm, ok := obj.(*corev1.ConfigMap); cm == nil || !ok {
			err = fmt.Errorf("Error: Unknown type")   1️⃣
		} else {
			err = fmt.Errorf("No billing data")       2️⃣
			if cm.Data != nil {
				if value, ok := cm.Data["data"]; ok {
					resp.CspConfigData = value
					var cspConfig tCspConfig
					if err = json.Unmarshal([]byte(value), &cspConfig); err == nil {   3️⃣
						if json_data, e := json.Marshal(cspConfig.Versions); e == nil {
							resp.AdapterVersions = string(json_data)
						}
						resp.CspErrors = cspConfig.Errors
						resp.CspConfigFrom = "local cluster"
						if tExpire, err = time.Parse("2006-01-02T15:04:05.000000-07:00", cspConfig.Expire); err == nil {

						}
					} else {
						log.WithFields(log.Fields{"err": err}).Error()
					}
				}
			}
		}
	}

	if obj, err = global.ORCH.GetResource(RscTypeConfigMap, NvAdmSvcNamespace, "metering-archive"); err == nil {  4️⃣
		if cm, ok := obj.(*corev1.ConfigMap); cm == nil || !ok {
			err = fmt.Errorf("Error: Unknown type")
		} else if cm.Data != nil {
			resp.MeteringArchiveData, _ = cm.Data["archive"]
		}
	}


	if err != nil || len(resp.CspErrors) > 0 {     5️⃣
		if err != nil { 
			resp.NvError = err.Error()
		}
		log.WithFields(log.Fields{"compliant": resp.Compliant, "nvError": resp.NvError, "expire": dataExpiredErr, "cspErrors": resp.CspErrors}).Error()
	}

	return resp
}

```